tests/test_gbdesAstrometricFit.py

```diff
@@ -1058,6 +1061,114 @@ def test_inputCameraModel(self):
+
+    def testProperMotion(self):
+        """Calculate the change in position due to proper motion for a given
+        time change, and compare the results against astropy.
+        """
+
+        parallaxes = np.zeros(len(self.RAs))
+        refEpoch = 57100
+
+        data = pd.DataFrame(
+            {
+                "ra": self.RAs,
+                "dec": self.Decs,
+                "pmRA": self.pmRAs,
+                "pmDec": self.pmDecs,
+                "parallax": parallaxes,
+                "MJD": self.mjds,
+            }
+        )
+        ra_correction, dec_correction = calculate_apparent_motion(data, refEpoch)
+
+        pmRA_cosDec = self.pmRAs * np.cos((self.Decs * u.degree).to(u.radian))
+        coords = SkyCoord(
+            ra=self.RAs * u.degree,
+            dec=self.Decs * u.degree,
+            pm_ra_cosdec=pmRA_cosDec * u.mas / u.yr,
+            pm_dec=self.pmDecs * u.mas / u.yr,
+            obstime=Time(refEpoch, format="mjd"),
+        )
+
+        newCoords = coords.apply_space_motion(new_obstime=Time(self.mjds, format="mjd"))
+        astropy_dRA = newCoords.ra.degree - coords.ra.degree
+        astropy_dDec = newCoords.dec.degree - coords.dec.degree
+
+        np.testing.assert_allclose(astropy_dRA, ra_correction.value, rtol=1e-6)
+        np.testing.assert_allclose(astropy_dDec, dec_correction.value, rtol=1e-6)
+
+    def testParallax(self):
+        """Calculate the change in position due to parallax for a given time
+        change and compare the results against astropy. Astropy will not give
+        the parallax part separate from other sources of space motion, such as
+        annual aberration, but it can be obtained by comparing the calculated
+        positions of an object with a given parallax and one with a much
+        further distance. The result is still only approximately the same, but
+        is close enough for accuracy needed here.
+        """
+        pms = np.zeros(len(self.RAs))
+        refEpoch = 57100
+
+        data = pd.DataFrame(
+            {
+                "ra": self.RAs,
+                "dec": self.Decs,
+                "pmRA": pms,
+                "pmDec": pms,
+                "parallax": self.parallaxes,
+                "MJD": self.mjds,
+            }
+        )
+        ra_correction, dec_correction = calculate_apparent_motion(data, refEpoch)
+
+        coords = SkyCoord(
+            ra=self.RAs * u.degree,
+            dec=self.Decs * u.degree,
+            pm_ra_cosdec=pms * u.mas / u.yr,
+            pm_dec=pms * u.mas / u.yr,
+            distance=Distance(parallax=self.parallaxes * u.mas),
+            obstime=Time(refEpoch, format="mjd"),
+        )
+        refCoords = SkyCoord(
+            ra=self.RAs * u.degree,
+            dec=self.Decs * u.degree,
+            pm_ra_cosdec=pms * u.mas / u.yr,
+            pm_dec=pms * u.mas / u.yr,
+            distance=1e10 * Distance(parallax=self.parallaxes * u.mas),
```

Comment:
Why 1e10?

Reply:
The point is to compare an object with a much more distant object at the same RA/Dec because parallax will only affect the position of the nearby object. This is described in the docstring, but I'll add a comment here.